Explicitly initialize color properties.

mdubet opened the pull request:
#### c00fe941b56e6b371bfc1e98d175059f269635f4
<pre>
Explicitly initialize color properties.
<a href="https://bugs.webkit.org/show_bug.cgi?id=245359">https://bugs.webkit.org/show_bug.cgi?id=245359</a>

Reviewed by Antti Koivisto.

Before this change, those properties were implicitely
default-initialized to empty color, which is
interpreted as current color.
This change makes the current color initialization
explicit.

* Source/WebCore/rendering/style/SVGRenderStyleDefs.cpp:
(WebCore::StyleFillData::StyleFillData):
* Source/WebCore/rendering/style/StyleRareInheritedData.cpp:
(WebCore::StyleRareInheritedData::StyleRareInheritedData):

Canonical link: <a href="https://commits.webkit.org/254627@main">https://commits.webkit.org/254627@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/51d1af5d06382573c3c6de72d0c74f213c308ac7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89688 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/34239 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20402 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98998 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/155809 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/93696 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32733 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/28207 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/82056 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/93386 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/95335 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/26011 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/76538 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25968 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80893 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/80765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68959 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/30487 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/14846 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/30237 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15781 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3250 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33685 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/38727 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/32396 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34873 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/rendering/style/StyleRareInheritedData.cpp:
```diff
@@ -81,6 +81,12 @@ StyleRareInheritedData::StyleRareInheritedData()
+    , visitedLinkTextEmphasisColor(RenderStyle::initialTextEmphasisColor())
+    , caretColor(RenderStyle::currentColor())
+    , visitedLinkCaretColor(RenderStyle::currentColor())
+    , accentColor(RenderStyle::currentColor())
```
This is wrong, and we should fix it in a different bug, since the initial value of `accent-color` is `auto`, not `currentColor`, but it doesn't matter too much since `hasAutoAccentColor` takes care of it anyway.

Anyway, not blocking on this, since this was just the previous behavior :)

mdubet replied:
Yep this is exactly the reason why I haven't create a specific named initializer for them (such as `initialAccentColor()`).
We should find a proper way to deal with `auto`.

mdubet committed:
Explicitly initialize color properties.
https://bugs.webkit.org/show_bug.cgi?id=245359

Reviewed by Antti Koivisto.

Before this change, those properties were implicitely
default-initialized to empty color, which is
interpreted as current color.
This change makes the current color initialization
explicit.

* Source/WebCore/rendering/style/SVGRenderStyleDefs.cpp:
(WebCore::StyleFillData::StyleFillData):
* Source/WebCore/rendering/style/StyleRareInheritedData.cpp:
(WebCore::StyleRareInheritedData::StyleRareInheritedData):

Canonical link: https://commits.webkit.org/254627@main
